### ISSUE TITLE
Explicit foreign keys can't reference tables with custom sql name

### ISSUE THREAD

**Issue by gmarpons** [yesodweb/persistent#373]
This works (see #41):

```
Tree sql=trees
  parent TreeId
```

But this, doesn't, neither referencing the table as `Tree` (migration error) nor as `trees` (TH compilation error):

```
Tree sql=trees
  name String
  parent String Maybe
  Primary name
  Foreign Tree fkparent parent
```

**Comment by gregwebs** [comment]
Which backend are you using? This seems to work on PostgreSQL for me now (there are some recent changes that could have fixed it). I added it as a test case: https://github.com/yesodweb/persistent/commit/c5ef27792eaa0d7dd741dcf436f71aa2faacd76d

**Comment by gmarpons** [comment]
Hi Greg. I use PostgreSQL backend. Not working for me, yet.

I've tried with revision 73c4cd6a35217224a21fe32553501d29667e8b81 of persistent. What I get when I execute `migrateAll`:

```
Migrating: CREATe TABLE "trees"( PRIMARY KEY ("name"),"name" VARCHAR NOT NULL,"parent" VARCHAR NULL)
[Debug#SQL] CREATe TABLE "trees"( PRIMARY KEY ("name"),"name" VARCHAR NOT NULL,"parent" VARCHAR NULL); [] @(<unknown>:<unknown> <unknown>:0:0)
Migrating: ALTER TABLE "trees" ADD CONSTRAINT "treefkparent" FOREIGN KEY("parent") REFERENCES "tree"("name")
[Debug#SQL] ALTER TABLE "trees" ADD CONSTRAINT "treefkparent" FOREIGN KEY("parent") REFERENCES "tree"("name"); [] @(<unknown>:<unknown> <unknown>:0:0)
*** Exception: SqlError {sqlState = "42P01", sqlExecStatus = FatalError, sqlErrorMsg = "relation \"tree\" does not exist", sqlErrorDetail = "", sqlErrorHint = ""}
```

So, when adding the foreign key constraint, the table name used is still `tree` instead of `trees`.

**Comment by winterland1989** [comment]
The test case didn't rename the table, try anything other than `tree` will fail the test.

**Comment by Philonous** [comment]
I'm being bitten by this too. Has there been any progress on this issue?

**Comment by jonkri** [comment]
+1

**Comment by MHova** [comment]
+1

**Comment by parsonsmatt** [comment]
Fixed and released in `persistent-2.10.5.2`
